src/nlp/arrow_dataset.py

```diff
@@ -903,6 +903,7 @@ def select(
         load_from_cache_file: bool = True,
         cache_file_name: Optional[str] = None,
         writer_batch_size: Optional[int] = 1000,
+        reader_batch_size: Optional[int] = 1000,
```

Comment:
Could you add it to the docstrings ?

Reply:
Also I would rename it `batch_size` (like in `map`, `filter`, etc.)
What do you think ?

Reply:
I thought about that, but I didn't want this option to conflict with the `batch_size` present in `map` and `filter`. In those functions, the `batch_size` is something that changes the user-visible behavior of `map` (it determines how many examples are sent to my callback per call.)

This `reader_batch_size` is purely behind the scenes and doesn't influence the output in a way that the end-user can see. So I thought it would be less confusing if they had different name. If `select` had a `batch_size`, I might expect it to return items in a list of chunks or something like that.

No strong opinions though.

Reply:
Yea that makes sense. Let's keep `reader_batch_size`.